Holtron.Net/Network/Encryption/NetEncryptionAESGCM.cs

```diff
@@ -0,0 +1,82 @@
+using System.Security.Cryptography;
+
+namespace Holtron.Net.Network.Encryption
+{
+    public class NetEncryptionAESGCM : INetEncryption
+    {
+        // This implementation is from
+        // https://stackoverflow.com/questions/60889345/using-the-aesgcm-class
+        private readonly AesGcm aesGcm;
+
+        public NetEncryptionAESGCM(byte[] key)
+        {
+            var derivedKey = new Rfc2898DeriveBytes(key, new byte[8], 1000).GetBytes(16);
+            aesGcm = new AesGcm(derivedKey);
+        }
+
+        public NetEncryptionAESGCM(string key)
+        {
+            var derivedKey = new Rfc2898DeriveBytes(key, new byte[8], 1000).GetBytes(16);
+            aesGcm = new AesGcm(derivedKey);
+        }
+
+        public NetEncryptionAESGCM(Span<byte> key)
```

Comment:
You could probably chain this constructor into the one that takes `byte[]` for the key:

```csharp
public NetEncryptionAESGCM(Span<byte> key) :
    this(key.ToArray())
{
}
```

Reply:
Chained it and another constructor to the first.